commands/root.go

```diff
@@ -16,55 +16,131 @@
 package commands
 
 import (
+	"context"
 	"fmt"
 	"log"
 	"os"
 	"strings"
 
+	"github.com/c-bata/go-prompt"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
 	clustercmd "github.com/hazelcast/hazelcast-commandline-client/commands/cluster"
 	mapcmd "github.com/hazelcast/hazelcast-commandline-client/commands/types/map"
 	"github.com/hazelcast/hazelcast-commandline-client/internal"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/cobraprompt"
 )
 
 var (
-	cfgFile   string
-	addresses string
-	cluster   string
-	token     string
-	rootCmd   = &cobra.Command{
+	RootCmd = &cobra.Command{
 		Use:   "hzc {cluster | help | map} [--address address | --cloud-token token | --cluster-name name | --config config]",
 		Short: "Hazelcast command-line client",
 		Long:  "Hazelcast command-line client connects your command-line to a Hazelcast cluster",
 		Example: "`hzc map --name my-map put --key hello --value world` - put entry into map directly\n" +
 			"`hzc help` - print help",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if _, err := internal.MakeConfig(); err != nil {
+				return err
+			}
 			return cmd.Help()
 		},
 	}
 )
 
+func addressAndClusterNamePrefix() (prefix string, useLive bool) {
+	return fmt.Sprintf("hzc %s@%s> ", internal.Configuration.Cluster.Network.Addresses[0], internal.Configuration.Cluster.Name), true
+}
+
+var advancedPrompt = &cobraprompt.CobraPrompt{
+	RootCmd:                  RootCmd,
+	PersistFlagValues:        true,
+	ShowHelpCommandAndFlags:  true,
+	ShowHiddenFlags:          true,
+	SuggestFlagsWithoutDash:  true,
+	DisableCompletionCommand: true,
+	AddDefaultExitCommand:    true,
+	GoPromptOptions: []prompt.Option{
+		prompt.OptionTitle("hazelcast commandline client"),
```

Comment:
Is this title displayed as the title of the terminal window? How about: "Hazelcast Commandline Client" ?

Reply:
Yes it is, sure that makes sense. I would also prefer a shorter name since there are no guarantees that we will have sufficient space to display the whole title. What do you think of "Hazelcast Cli"?

Reply:
How about "Hazelcast Client"?